src/auditor/auditor.py

```diff
@@ -81,6 +80,55 @@ def log_report(report_dict, report_file, separator='|', rotate_count=18):
         report_logger.info(separator.join(item_list))
 
 
+class Metatable:
+    """
+    Represents the metatable containing information about SGID items uploaded to AGOL.
+    read_metatable() can be called on both the SGID AGOLItems table or the AGOL-hosted AGOLItems_Shelved table.
+    Table stored as self.metatable_dict in the following format:
+        {item_id: [table_sgid_name, table_agol_name, table_category, table_authoritative]}
+    Any duplicate item ids (either a table has the same AGOL item in more than one row, or the item id exists in
+    multiple tables) are added to the self.duplicate_keys list.
+    """
+
+    def __init__(self):
+        #: A dictionary of the metatable records, indexed by the metatable's itemid
+        #: values: {item_id: [table_sgid_name, table_agol_name, table_category, table_authoritative]}
+        self.metatable_dict = {}
+        self.duplicate_keys = []
+
+    def read_metatable(self, table, fields):
+        """
+        Read metatable 'table' into self.metatable_dict.
+
+        Returns: list of any duplicate AGOL item ids found
+        """
+
+        with arcpy.da.SearchCursor(table, fields) as meta_cursor:
+            for row in meta_cursor:
+
+                if fields[-1] == 'Authoritative':  #: SGID's AGOLItems table's last field is "Authoritative"
```

Comment:
**suggestion**: Relying on field ordering is a little sketchy in my opinion. Better to check if the field is present in a list or pass an `is_sgid` boolean parameter or something.